website/blog/2019-01-22-jest-24-refreshing-polished-typescript-friendly.md

```diff
@@ -0,0 +1,92 @@
+
+The second one is an issue where Jest runs out of memory if the difference in serialization of expected and received value has a huge number of insertion changes (either unexpected because of mistake in test or defect in serializer or expected because of temporary failures during test-driven development). [@ittordepam](https://twitter.com/ittordepam) has [replaced](https://github.com/facebook/jest/pull/6961) the previous diffing algorithm with `diff-sequences` package, which should fix this issue because it uses the theoretical minimum amount of memory. It also opens up for word-diffing in the future.
+
+## Other highlights
+
+- Jest already provides a module called `jest-worker` which helps you parallelize work across CPUs. Node 10 came with an experimental module called `worker_threads`, which is similar to Worker threads in the browser. This was un-flagged in Node's latest 11.7.0 release, and `jest-worker` will automatically use that if available instead of spawning new processes via `child-process`. This makes it even faster!
+- We have some improvements for `globalSetup` and `globalTeardown` as well - code transformation will be applied to them similar to `setupFiles` and they are now supported as part of `projects`.
+- You can [configure](https://github.com/facebook/jest/pull/6143) Jest's snapshot location, this is mainly useful if you are building tools which use Jest in a larger build process.
+- A quirk of Jest's CLI has been that while some flags and options have been camel cased (such as `runInBand`), others have not been (such as `no-cache`). In Jest 24, both are recognized, meaning you can write your CLI arguments however you want.
+- We've renamed `setupTestFrameworkScriptFile` to `setupFilesAfterEnv`, and made it into an array. We hope this will make it more obvious what the options is for. We have plans to further overhaul the configuration in the next major, see the paragraph in the section below.
+- To reduce the amount of magic Jest performs to “just work™”, in this release we decided to drop automatic injection of `regenerator-runtime` for async code to work. It's not always necessary (e.g. in Node 8+) and we believe it's the user's responsibility to handle it, which is already done if you use `@babel/preset-env` and `@babel/runtime`.
+- Lastly, we've upgraded to Micromatch 3. While this might not affect every user, it is stricter in its parsing of globs than version 2, which is used in Jest 23. Please read through [this](https://github.com/micromatch/micromatch/issues/133#issuecomment-404211484) and linked issues for examples of invalid globs in case you have problems.
+
+## The future
+
+We are incredibly humbled by the results in [State Of JS 2018](https://2018.stateofjs.com/awards/), where Jest won the “Highest Satisfaction” award. Another huge thing to happen in 2018 was in October, when Jest passed 2 million weekly downloads for the first time, actually passing the business stalwart that is Mocha for the first time.
```

Comment:
I don't think it's productive to mention this type of comparison. Can we celebrate the number of downloads of Jest without diminishing the value of other testing frameworks in the same sentence?

Reply:
"Another huge thing to happen in 2018 was in October, when Jest passed 2 million weekly downloads for the first time. Thank you."

Reply:
good call